### PR TITLE
feat: support additional and anyof/oneof field

### PR DESCRIPTION
**Pull request by Ricbet**
### Types

- [x] 🎉 New Features

### <samp>🤖 Generated by Copilot at 169772e</samp>

*  Add the CheckboxWidget component to render boolean schemas with a checkbox input and a label and description ([link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-21384e6a7fb0d7a1661f8d7d9f822845611f433a2c2dedaf9cc6b4d1da3724dbR1-R78), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-6771b64600fe1c461bcb549a971144b394bd7bf1ada950c86e4b2cdeb150e805R19-R36), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bR204))
*  Modify the SelectWidget and TextWidget components to render the title and description of the schema using the TitleFieldTemplate and DescriptionFieldTemplate components ([link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-f86d1cd15965214231cce3fa2b44d0fa407e9832d41b5ba1afb1847ddabe6ebbL1-R9), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-f86d1cd15965214231cce3fa2b44d0fa407e9832d41b5ba1afb1847ddabe6ebbL8-R28), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-f86d1cd15965214231cce3fa2b44d0fa407e9832d41b5ba1afb1847ddabe6ebbL19-R66), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-22e325d5312d9c44550fad5f1619256919014d486f96e39dcb76c98c1ac799f3L1-R10), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-22e325d5312d9c44550fad5f1619256919014d486f96e39dcb76c98c1ac799f3L23-R62), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-22e325d5312d9c44550fad5f1619256919014d486f96e39dcb76c98c1ac799f3L55-R131))
*  Add the TitleFieldTemplate and DescriptionFieldTemplate components to render the title and description of the schema with a label and a markdown component ([link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-f560f59e8908590c9e2a034725b8fef61e88b906398360d2773c35c0bdc27c46R1-R35), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-6e7f7e9f5e5e44e10732f9da7583ce43aa515c3a938f4b7fdd379e2dc7601c9bL2-R5), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-6e7f7e9f5e5e44e10732f9da7583ce43aa515c3a938f4b7fdd379e2dc7601c9bL13-R20))
*  Add the WrapIfAdditionalTemplate component to wrap the field with a label and description if the schema has additional properties or the additionalProperty flag ([link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-86b8f1061b6e5516b2cd0d79f787c704e7493273a02fe0c49a82856103a26bb0R1-R136), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-483d2e76d3cdfee74cfb9ce43e708f0814fd6e27ab8b431fbd6414944c69105dL1-R71))
*  Add the BaseInputTemplate component to render a base input element for string, number, integer, or boolean schemas ([link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-62966662e292f6274876f75f7069345f26be1a533ade1f19bd35c5da33a5d1c4R1-R79), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bR299-R301))
*  Modify the LaunchBody component to modify the schema properties before passing them to the Form component, and to add the OneOfField, CheckboxWidget, and the new templates to the Form component props ([link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bL28-R37), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bR263-R272), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bL272-R292), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bR299-R301))
*  Modify the AnyOfField component to import the TitleField and DescriptionField components and the IJSONSchema interface ([link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-94e917ae3faff1c340391acb24ecf29a47c7a33498630923496345b20c575a5fL1-R89))
*  Modify the SubmitButton component to customize the appearance and text of the button ([link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-44a1e5b7f9df9d26b77afac1161fdfd7150ea048277ed91f0a55763512916967R8-R9), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-44a1e5b7f9df9d26b77afac1161fdfd7150ea048277ed91f0a55763512916967L38-R44))
*  Modify the FieldTemplate component to use the WrapIfAdditionalTemplate component ([link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-483d2e76d3cdfee74cfb9ce43e708f0814fd6e27ab8b431fbd6414944c69105dL1-R71))
*  Modify the ObjectFieldTemplate component to wrap the object title, description, and properties with a container_field div element, and to add an add button for the additional properties if the schema can be expanded ([link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-293c8af0ffda57fcd1e7c10e376a2886b6299fbe7d91deb179215731ea40d3deL10-R16), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-293c8af0ffda57fcd1e7c10e376a2886b6299fbe7d91deb179215731ea40d3deL18-R35), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-293c8af0ffda57fcd1e7c10e376a2886b6299fbe7d91deb179215731ea40d3deL26-R91))
*  Modify the styles for the object fields and the additional properties in the `json-templates.module.less` file ([link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-0b47ea19c9728a74dab1d95d912bbfb2059e790bb2eeb5d49963010a73ee4b2dL21-L26), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-0b47ea19c9728a74dab1d95d912bbfb2059e790bb2eeb5d49963010a73ee4b2dL38-R43), [link](https://github.com/opensumi/core/pull/2692/files?diff=unified&w=0#diff-0b47ea19c9728a74dab1d95d912bbfb2059e790bb2eeb5d49963010a73ee4b2dL56-R99))

image
image
image

- [x] 新增 checkbox widget 部件
- [x] 支持渲染 markdownDescription
- [x] 新增 additional template，用于处理 additionalProperties 属性
- [x] 优化代码

### <samp>🤖 Generated by Copilot at 169772e</samp>

This pull request improves the debug preferences form by adding support for more schema features and widgets, such as titles, descriptions, additional properties, and checkboxes. It also updates the components and templates that render the schema form, and adds some styles and dependencies. It mainly affects the files in the `packages/debug/src/browser/preferences` folder.

### CODE REVIEW

File: packages/debug/src/browser/preferences/templates/additional-template.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          ) : (
+            <div className={styles.form_additional_container}>
+              <div className={styles.form_additional}>
+                {properties &&
+                  Object.keys(properties).map((item) => (
+                    <div className={styles.form_group}>
+                      <Input
+                        className={styles.form_control}
+                        defaultValue={label}
+                        placeholder={'请输入 Key'}
```
Comment (erha19) [inline packages/debug/src/browser/preferences/templates/additional-template.tsx:117]: placeholder 这里做一下 localize ?

File: packages/debug/src/browser/preferences/templates/button-template.tsx [hunk 2]
```diff
@@ -35,4 +37,8 @@ export const CopyButton = (props: IconButtonProps) => (
   </Button>
 );
 
-export const SubmitButton = (props: IconButtonProps) => <Button>{localize('ButtonOK')}</Button>;
+export const SubmitButton = (props: IconButtonProps) => (
+  <Button {...props} type='secondary' icon={defaultIconfont.plus} className={styles.add_new_field}>
+    <span className={getIcon(defaultIconfont.plus)}></span> 新增配置项
```
Comment (erha19) [inline packages/debug/src/browser/preferences/templates/button-template.tsx:42]: 这里也是

**Review by erha19 — CHANGES_REQUESTED** [review]

File: packages/debug/src/browser/preferences/templates/button-template.tsx (hunk 2 above)
Comment (Ricbet) [reply packages/debug/src/browser/preferences/templates/button-template.tsx]: done

File: packages/debug/src/browser/preferences/templates/json-templates.module.less [hunk 3]
```diff
@@ -35,16 +29,18 @@
...
-  }
+  .object_field_container {
+    .object_title {
+      margin-bottom: 16px;
+    }
 
-  .object_description {
-    margin-bottom: 8px;
-  }
+    .object_description {
```
Comment (erha19) [inline packages/debug/src/browser/preferences/templates/json-templates.module.less:37]: 这里缩进好像有点问题？

File: packages/debug/src/browser/preferences/components/checkbox-widget.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    uiSchema,
+  } = props;
+  const { readonlyAsDisabled = true } = formContext as GenericObjectType;
+  const DescriptionFieldTemplate = getTemplate<'DescriptionFieldTemplate', T, S, F>(
+    'DescriptionFieldTemplate',
+    registry,
+    options,
+  );
+
+  const handleChange = ({ target }: any) => onChange(target.checked);
```
Comment (erha19) [inline packages/debug/src/browser/preferences/components/checkbox-widget.tsx:45]: 重复创建的函数方法是不是用 useCallback 包裹好一点？

File: packages/debug/src/browser/preferences/components/fields/any-of-field.tsx [hunk 5]
```diff
@@ -1,32 +1,90 @@
...
+    schema,
+    uiOptions,
+  } = props;
+  const { readonlyAsDisabled = true } = formContext as GenericObjectType;
 
   const handleTextChange = ({ target }: React.ChangeEvent<HTMLInputElement>) =>
     onChange(target.value === '' ? options.emptyValue : target.value);
 
-  const handleBlur = ({ target }: React.FocusEvent<HTMLInputElement>) => onBlur(id, target.value);
+  const handleBlur = ({ target }: React.FocusEvent<HTMLInputElement>) => onBlur(id!, target.value);
```
Comment (erha19) [inline packages/debug/src/browser/preferences/components/fields/any-of-field.tsx:44]: 很多地方都缺少 useCallback 包裹，直接引用 Ref 性能好一点

**Review by erha19 — CHANGES_REQUESTED** [review]

File: packages/debug/src/browser/preferences/templates/json-templates.module.less (hunk 3 above)
Comment (Ricbet) [reply packages/debug/src/browser/preferences/templates/json-templates.module.less:37]: 这里缩进没问题，因为多了一层嵌套

File: packages/debug/src/browser/preferences/components/checkbox-widget.tsx (hunk 4 above)
Comment (Ricbet) [reply packages/debug/src/browser/preferences/components/checkbox-widget.tsx:45]: done

**Review by erha19 — APPROVED** [review]

**Pull request merged**